app/src/main/java/com/example/playlistmakernewversion/SearchActivity.kt

```diff
@@ -51,11 +89,77 @@ class SearchActivity : AppCompatActivity() {
         }
         inputEditText.addTextChangedListener(simpleTextWatcher)
 
-        val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
-        recyclerView.adapter = TrackAdapter(arrayTracks)
-        recyclerView.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
+        recyclerView = findViewById(R.id.recyclerView)
+        recyclerView.adapter = tracksAdapter
+        recyclerView.layoutManager = LinearLayoutManager(
+            this,
+            LinearLayoutManager.VERTICAL,
+            false
+        )
+
+        inputEditText.setOnEditorActionListener { _, actionId, _ ->
+            if (actionId == EditorInfo.IME_ACTION_DONE) {
+                if (inputEditText.text.isNotEmpty()) {
+                    placeHolderNothingFound.visibility = View.GONE
+                    placeholderErrorNetwork.visibility = View.GONE
+                    itunesService.search(inputEditText.text.toString()).enqueue(object :
+                        Callback<TracksResponse> {
+                        override fun onResponse(
+                            call: Call<TracksResponse>,
+                            response: Response<TracksResponse>
+                        ) {
+                            if (response.code() == 200) {
```

Comment:
```suggestion
                            if (response.isSuccessfull) {
```
Лучше написать так, чтобы не хардкодить коды ответов http запросов